src/main/java/seedu/address/logic/commands/SearchAnimalCommand.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliAnimalSyntax.BREED;
+import static seedu.address.logic.parser.CliAnimalSyntax.DATE_OF_ADMISSION;
+import static seedu.address.logic.parser.CliAnimalSyntax.DATE_OF_BIRTH;
+import static seedu.address.logic.parser.CliAnimalSyntax.NAME;
+import static seedu.address.logic.parser.CliAnimalSyntax.PET_ID;
+import static seedu.address.logic.parser.CliAnimalSyntax.SEX;
+import static seedu.address.logic.parser.CliAnimalSyntax.SPECIES;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.AnimalModel;
+import seedu.address.model.animal.KeywordPredicate;
+
+/**
+ * Searches for animals in the catalog based on keyword given (case-insensitive).
+ */
+public class SearchAnimalCommand extends AnimalCommand {
+    public static final String COMMAND_WORD = "search";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Searches for animals in the catalog based on keyword given (case insensitive). "
+            + "Parameters: KEYWORD\n"
+            + "Example: " + COMMAND_WORD + " Pookie";
```

Comment:
The example given `COMMAND_WORD + "Pookie"`. I assumed this is an example of searching the Animals by their `Name` attribute. If so, is the `n/` prefix required, or is searching by name the default behaviour?